### PR TITLE
Fix Custom PR Title in Aggregated Mode

### PR DESCRIPTION
**Pull request by EyalDelarea**
- [x] All [tests](https://github.com/jfrog/frogbot#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
---

**Bug Description:**
When aggregate mode is on, custom PR title template wasn't used.

**Fix:**
Now custom templates will be loaded into aggregated PR title with tech suffix.
[Custom template] + "Go Dependencies".

Also - If more then one technology is detected in the same project, append to aggregate fixes.
[Custom template] + "Go,Npm Dependencies". for exmaple.

image

### CODE REVIEW

File: utils/git.go [hunk 1]
```diff
@@ -431,7 +450,43 @@ func setGoGitCustomClient() {
...
+	if len(techsArray) < 2 {
+		return techsArray[0].ToFormal()
+	}
+	// Append each tech with a separator
+	for index, tech := range techsArray {
+		result += tech.ToFormal()
+		if index != len(techsArray)-1 {
+			result += ","
+		}
+	}
```
Comment (sverdlov93) [inline utils/git.go:486]: strings.join( techs, ",")

File: scanrepository/scanrepository.go [hunk 2]
```diff
@@ -40,7 +40,7 @@ type ScanRepositoryCmd struct {
 	// Determines whether to open a pull request for each vulnerability fix or to aggregate all fixes into one pull request
 	aggregateFixes bool
 	// The current project technology
-	projectTech coreutils.Technology
+	projectTech []coreutils.Technology
```
Comment (omerzi) [inline scanrepository/scanrepository.go:43]: A Project represents a technology with its working dirs, installCommand, etc. So this change is unnecessary.

File: utils/git.go [hunk 3]
```diff
@@ -323,10 +323,11 @@ func (gm *GitManager) GenerateCommitMessage(impactedPackage string, fixVersion s
 	return formatStringWithPlaceHolders(template, impactedPackage, fixVersion, "", true)
 }
 
-func (gm *GitManager) GenerateAggregatedCommitMessage(tech coreutils.Technology) string {
+func (gm *GitManager) GenerateAggregatedCommitMessage(tech []coreutils.Technology) string {
 	template := gm.customTemplates.commitMessageTemplate
 	if template == "" {
-		template = outputwriter.GetAggregatedPullRequestTitle(tech)
+		// Aggregate commit message could not include each package, use PR title.
```
Comment (omerzi) [inline utils/git.go:329]: Unclear comment

File: utils/git.go [hunk 4]
```diff
@@ -431,7 +450,40 @@ func setGoGitCustomClient() {
...
+		return techsArray[0].ToFormal()
+	}
+	var techString []string
+	for _, tech := range techsArray {
+		techString = append(techString, tech.ToFormal())
+	}
+	return strings.Join(techString, ",")
+}
+
+func removeMiddleSpaces(text string) string {
```
Comment (omerzi) [inline utils/git.go:487]:
```suggestion
// Normalizes whitespace in text, ensuring that words are separated by a single space, and any extra whitespace is removed.
func normalizeWhitespaces(text string) string {
```

File: utils/git.go [hunk 5]
```diff
@@ -431,7 +450,40 @@ func setGoGitCustomClient() {
...
+	return strings.Join(strings.Fields(text), " ")
```
Comment (omerzi) [inline utils/git.go:488]: Add unit test

**Review by omerzi — CHANGES_REQUESTED** [review]

File: utils/git.go (hunk 4 above)
Comment (omerzi) [reply utils/git.go:487]: + move it to utils

**Pull request merged**